File: app/src/main/java/com/example/madarsofttask/ui/list/UserAdapter.java
package com.example.madarsofttask.ui.list;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.madarsofttask.R;
import com.example.madarsofttask.data.models.User;

import java.util.ArrayList;
import java.util.List;

public class UserAdapter extends RecyclerView.Adapter<UserAdapter.UserHolder> {
    private List<User> users = new ArrayList<>();

    @NonNull
    @Override
    public UserHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_layout, parent, false);
        return new UserHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull UserHolder holder, int position) {
        User currentUser = users.get(position);
        holder.bind(currentUser);
    }

    @Override
    public int getItemCount() {
        return users.size();
    }

    void setUsers(List<User> users) {
        this.users = users;
        notifyDataSetChanged();
    }

    static class UserHolder extends RecyclerView.ViewHolder {
        private TextView textViewTitle;
        private TextView textViewName;
        private TextView textViewAge;
        private TextView textViewGender;

        UserHolder(View itemView) {
            super(itemView);
            textViewTitle = itemView.findViewById(R.id.title_textView);
            textViewName = itemView.findViewById(R.id.name_textView);
            textViewAge = itemView.findViewById(R.id.age_textView);
            textViewGender = itemView.findViewById(R.id.gender_textView);
        }

        void bind(User user) {
            textViewTitle.setText(user.getJobTitle());
            textViewName.setText(user.getName());
            textViewAge.setText(String.valueOf(user.getAge()));
            textViewGender.setText(user.getGender());
        }
    }
}

File: app/src/main/java/com/example/madarsofttask/ui/list/UserAdapter.kt
package com.example.madarsofttask.ui.list

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.example.madarsofttask.R
import com.example.madarsofttask.data.models.User
import com.example.madarsofttask.ui.list.UserAdapter.UserHolder
import java.util.*
import kotlinx.android.synthetic.main.item_layout.view.*

class UserAdapter : RecyclerView.Adapter<UserHolder>() {
    private var users: List<User> = ArrayList()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.item_layout, parent, false)
        return UserHolder(itemView)
    }

    override fun onBindViewHolder(holder: UserHolder, position: Int) {
        val currentUser = users[position]
        holder.bind(currentUser)
    }

    override fun getItemCount(): Int {
        return users.size
    }

    fun setUsers(users: List<User>) {
        this.users = users
        notifyDataSetChanged()
    }

    class UserHolder(itemView: View) : ViewHolder(itemView) {
        fun bind(user: User) {
            itemView.title_textView.text = user.jobTitle
    itemView.name_textView.text = user.name
            itemView.age_textView.text = user.age.toString()
    itemView.gender_textView.text=user.gender
        }

    }
}